pandas/core/groupby/generic.py

```diff
@@ -1308,22 +1308,10 @@ def _transform_general(self, func, *args, **kwargs):
         gen = self.grouper.get_iterator(obj, axis=self.axis)
         fast_path, slow_path = self._define_paths(func, *args, **kwargs)
 
-        for name, group in gen:
-            if group.size == 0:
-                continue
-            object.__setattr__(group, "name", name)
-
-            # Try slow path and fast path.
-            try:
-                path, res = self._choose_path(fast_path, slow_path, group)
-            except TypeError:
-                return self._transform_item_by_item(obj, fast_path)
-            except ValueError as err:
-                msg = "transform must return a scalar value for each group"
-                raise ValueError(msg) from err
-
+        def process_result(
```

Comment:
can this be a module level function?

Reply:
Yes - can either make it a module-level by passing self.obj or a class method. Do you see a reason to prefer one or the other?

Reply:
i think prefer a module level